### ISSUE TITLE
Error supression

### ISSUE THREAD

**Issue by drmax24** [getsentry/sentry-php#661]
Sentry 1.7 and 1.9.2 triggers program stop when it gets suppressed error like this:

$x = @$lastCredit['quantity'];

Undefined index: quantity

php 5.6.30

Can you make it work? 
Or how should i configure sentry to ignore supressed errors?

I can write
$x = (isset($lastCredit['quantity']) && $lastCredit['quantity'])?$lastCredit['quantity']:null;
But that is not the point of my question

**Comment by Jean85** [comment]
Under PHP 7 you would be able to write `$lastCredit['quantity'] ?? null`, but still, that's not the point. To avoid capturing those, you would have to raise the level at which the client captures, using the `error_types` option; that has to be configured in the same way as in the PHP ini, with error constants: https://secure.php.net/manual/en/errorfunc.constants.php

**Comment by drmax24** [comment]
I want to capture all errors except suppressed ones

**Comment by ste93cry** [comment]
>`$x = @$lastCredit['quantity'];`

This code is really bad, you should check that the key exists in the array map instead of just silencing the possible error if it doesn't. Said this, the code of the error handler is always a lot tricky but as far as I remember we catch all errors, regardless of the fact that they are silenced. You could anyway create and register your own error handler that forwards only certain errors to the Sentry error handler

**Comment by stayallive** [comment]
How do you mean it triggers a program stop? It shouldn't. An undefined index just throws a notice and php itself does not halt it either.

```php
<?php

require __DIR__ . '/vendor/autoload.php';

$client = (new Raven_Client('https://xxx@sentry.io/xxx'))->install();

$array = [];

$pizza = @$array['undefined'];

echo 'Hello World!';
```

This outputs the following, no halting just a Hello World!

```
# php56 index.php
Hello World!%   
```

It also never sends the notice to Sentry so something is going on... 

Are you using a framework (laravel/symfony)? If not can you post your Sentry installation code? And what options you are setting on the client?

**Comment by ste93cry** [comment]
This has been fixed in `2.0.1`. A new option has been added to configure whether suppressed errors should be captured by Sentry and by default this is `off`. Closing this issue since it's pretty old, feel free to reopen if you still have problems
